### PR TITLE
fix #416 add dockerfile and update docker docs

### LINKED ISSUE #416: Configuring TRD w/ Docker (instructions unclear)
**Describe the bug**
Unable to launch the TRD configuration wizard under the Tezos Docker environment. Tezos-Signer is running as a docker container and is able to send transactions out from the node.

**To Reproduce**
We have tried the following variations of executing the configure.py script, but have been unsuccessful in the Docker environment.
- python3 src/configure.py
- python3 src/configure.py -d
- python3 src/configure.py -d -E ../../mainnet.sh
- python3 src/configure.py -d -E http://0.0.0.0:9732
- python3 src/configure.py -d -E http://127.0.0.1:9732

For these commands, we have been receiving some variation of the following error:
```
2021-04-14 14:06:17,116 - MainThread - INFO - Arguments Configuration = {
 "reward_data_provider": "prpc",
 "network": "MAINNET",
 "reports_base": "~/pymnt/reports",
 "config_dir": "~/pymnt/cfg",
 "node_endpoint": "http://127.0.0.1:8732",
 "signer_endpoint": "http://127.0.0.1:6732",
 "docker": true,
 "verbose": "on",
 "dry_run": false,
 "api_base_url": null,
 "log_file": "logs/app.log"
}
2021-04-14 14:06:17,120 - MainThread - DEBUG - Error, request ->http://127.0.0.1:8732/chains/main/blocks/head/context/constants<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=8732): Max retries exceeded with url: /chains/main/blocks/head/context/constants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2901efc40>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:17,121 - MainThread - DEBUG - Error, request ->http://127.0.0.1:8732/chains/main/blocks/head/context/constants<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=8732): Max retries exceeded with url: /chains/main/blocks/head/context/constants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff29023e6a0>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:17,122 - MainThread - DEBUG - Failed to get network configuration constants from a local node (http://127.0.0.1:8732).
2021-04-14 14:06:17,779 - MainThread - DEBUG - Network configuration constants successfully loaded from a public node (https://mainnet-tezos.giganode.io).
2021-04-14 14:06:17,779 - MainThread - DEBUG - Network config {'NB_FREEZE_CYCLE': 5, 'BLOCK_TIME_IN_SEC': 60, 'BLOCKS_PER_CYCLE': 4096, 'BLOCKS_PER_ROLL_SNAPSHOT': 256, 'BLOCK_REWARD': 40000000, 'ENDORSEMENT_REWARD': 1250000, 'NAME': 'MAINNET'}
2021-04-14 14:06:40,289 - MainThread - DEBUG - RpcBlockApiImpl - node_url http://127.0.0.1:8732
2021-04-14 14:06:41,524 - MainThread - DEBUG - Error, request ->http://127.0.0.1:6732/keys/tz1PeyZkxMBgbVi<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=6732): Max retries exceeded with url: /keys/tz1PeyZkxMBgbVi (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff29024b2e0>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:41,525 - MainThread - DEBUG - Error, request ->http://127.0.0.1:6732/keys/tz1PeyZkxMBgbVi<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=6732): Max retries exceeded with url: /keys/tz1PeyZkxMBgbVi (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2901e0f10>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
```

**Screenshots**
```
t3zos@tezos-node:~/tezos-docker$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED       STATUS       PORTS                              NAMES
fbc3e14b5e4d   tezos-mainnet-signer   "/usr/local/bin/entr…"   3 hours ago   Up 3 hours                                      focused_dijkstra
_<... other containers>_
8ef5e7e9e7a8   tezos/tezos:v8.2       "/usr/local/bin/entr…"   4 weeks ago   Up 4 days    8732/tcp, 0.0.0.0:9732->9732/tcp   mainnet_node_1
```

**Environment**
 - Docker running on Ubuntu OS
 - mainnet.sh file is found two folders up from the tezos-reward-distributor repo (../../mainnet.sh)

### PR DESCRIPTION
**Pull request by nicolasochem**

### CODE REVIEW

**Comment by nicolasochem** [comment]
@jdsika are you open to deploy TRD containers to docker hub? I can assist with github actions, but we should coordinate on docker hub account creation.

File: docs/docker.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  * `reports` folder containing payout reports
+
+Access to a signer endpoint and node endpoint are assumed in the host network.
+
+Here are the steps:
+
+1. Build the docker container
+  ::
+
+    docker build -t trd .
```
Comment (vkresch) [inline docs/docker.rst:20]: @jdsika is it fine to use `trd` as a tag or should we stick with `tezos-reward-distributor:latest`? If we deploy it on dockerhub users should be able to pull by `docker pull tezos-reward-distributor/tezos-reward-distributor:latest` right?

**Review by vkresch — COMMENTED** [review]
It looks fine to me. I tested to build the image and running the container according to the documentation. The next steps would be to create a user on docker hub with the name `tezos-reward-distributor` and adding the docker hub secrets `DOCKER_USER` and `DOCKER_PASSWORD` to be able to build and push the docker image. I'm not sure what the current `DOCKER` secret is. (see also [this](https://medium.com/platformer-blog/lets-publish-a-docker-image-to-docker-hub-using-a-github-action-f0b17e5cceb3) tutorial or [this](https://github.com/marketplace/actions/build-and-push-docker-images) on docker deployment via gh actions)

**Comment by nicolasochem** [comment]
The github action looks good to me. What are the next steps @jdsika ? Can we please get these builds up and running soon?

I think you added a github secret for your account `stakenow`, but I think we should create a new account `tezos-reward-distributor`, then have a secret `DOCKER_USER=tezos-reward-distributor` and `DOCKER_SECRET` as the password.

**Comment by vkresch** [comment]
PR https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/442 is merged. Thus closing this PR.

**Pull request closed without merging**